packages/nest-pg-notify/src/lib/pg-notify.types.ts

```diff
@@ -0,0 +1,46 @@
+import type { FactoryProvider, ModuleMetadata, Type } from '@nestjs/common';
+import type Redis from 'ioredis';
+import type { ClientConfig } from 'pg';
+import type { Options, Subscriber } from 'pg-listen';
+import type Redlock from 'redlock';
+
+export interface PgNotificationOptions {
+    eventIdKey?: string | ((payload: any) => string | number);
+    type: string | ((payload: any) => boolean);
+    lockExpiry?: number;
+    notificationProcessedExpiry?: number;
+}
+
+export interface PgChannelOptions {
+    name: string;
+}
+
+export type PgNotifyRedisOptions = ConstructorParameters<typeof Redis>;
+
+type PgNotifyRedlockOptions = ConstructorParameters<typeof Redlock>;
+
+export interface PgNotifyConfig {
+    pg:
+        | Subscriber
```

Comment:
A little cleaner to extract the type for `pg` and `redlock` into types for each custom union entry as well as a type for the union.

Reply:
ya that makes sense. Good call.